src/section/section_members.rs

```diff
@@ -0,0 +1,253 @@
+// Copyright 2020 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use super::{
+    elders_info::EldersInfo,
+    member_info::{AgeCounter, MemberInfo, MemberState, MIN_AGE_COUNTER},
+};
+use crate::{
+    id::{P2pNode, PublicId},
+    xor_space::{Prefix, XorName},
+};
+use itertools::Itertools;
+use std::{
+    cmp::Ordering,
+    collections::{btree_map::Entry, BTreeMap},
+    mem,
+    net::SocketAddr,
+};
+
+/// Container for storing information about members of our section.
+#[derive(Debug, Eq, Serialize, Deserialize)]
+pub struct SectionMembers {
+    members: BTreeMap<XorName, MemberInfo>,
+    // Number that gets incremented every time a node joins or leaves our section - that is, every
+    // time `members` changes.
+    section_version: u64,
```

Comment:
`section_version`
Thinking out loud here..
I think `section_` is redundant. I would maybe go for just `version`, it's the version of the `SectionMembers`, that is perfectly clear and evident.
Maybe we want to call it `member_version` globally to be absolutely clear though (sticking to one term).

Reply:
I think version will be dropped now in favour of just key as that is unique and all we need is uniqueness here.

Reply:
This is a different version though. This one tracks any change to the section, not just elders change. It's purpose is to break ties when selecting which node to promote in case there are multiple ones with the same age - the one with the lower version (which means the one that joined the section earlier) wins.

Reply:
Ahh, several sorts of versions. Mm, a good distinction between the versions would be good then (if both are to be kept anyway).

If it tracks any change, then I go back on the previous suggestion, then it seems ok with just `section_version`. The other type of version, which is for elders change, would instead be distinguished somehow (if it is to be kept?),

Reply:
For ties, I would do this. 
1. Use age_counter
2. If still tied then select xor closest to the previous node that left to create this gap (or similar, even just the lex smallest one)

Reply:
Later I think it will make sense for any promote to elder that we force a name change and give the node a range to get an address for. That way on a split the elders should split circa 50/50. Needs some discussion though

Reply:
> Use age_counter

1. I'm already using it, but there can still be ties.
2. I though we were going to remove age_counter and use only age (https://github.com/maidsafe/routing/issues/2078)

Reply:
Good point, in that case then selecting on some criteria we can compute and is not gamable will be the best. 
How about select the Adult to promote that is oldest Adult and on tie then the Adult from the tie that is xor closest to the current section key?

Reply:
OK, added an issue for this to be addressed in a different PR: https://github.com/maidsafe/routing/issues/2088